clr-k8s-examples/create_stack.sh

```diff
@@ -50,53 +58,150 @@ function cluster_init() {
 }
 
 function kata() {
-	# Install kata artifacts using kata-deploy
-	kubectl apply -f 8-kata/deploy/kata-rbac.yaml
-	kubectl apply -f 8-kata/deploy/kata-deploy.yaml
-	kubectl apply -f 8-kata/
+	KATA_VER=${1:-1.8.0-kernel-config}
+	KATA_URL="https://github.com/kata-containers/packaging.git"
+	KATA_DIR="8-kata"
+	get_repo "${KATA_URL}" "${KATA_DIR}/overlays/${KATA_VER}"
+	set_repo_version "${KATA_VER}" "${KATA_DIR}/overlays/${KATA_VER}/packaging"
+	kubectl apply -k "${KATA_DIR}/overlays/${KATA_VER}"
+
 }
 
 function cni() {
-	kubectl apply -f 0-canal/rbac.yaml
-	kubectl apply -f 0-canal/canal.yaml
+	# note version is not semver
+	CANAL_VER=${1:-v3.3}
+	CANAL_URL="https://docs.projectcalico.org/$CANAL_VER/getting-started/kubernetes/installation/hosted/canal"
+	CANAL_DIR="0-canal"
+
+	# canal manifests are not kept in repo but in docs site so use curl
+	mkdir -p "${CANAL_DIR}/overlays/${CANAL_VER}/canal"
+	curl -o "${CANAL_DIR}/overlays/${CANAL_VER}/canal/canal.yaml" "$CANAL_URL/canal.yaml"
+	curl -o "${CANAL_DIR}/overlays/${CANAL_VER}/canal/rbac.yaml" "$CANAL_URL/rbac.yaml"
+	# canal doesnt pass kustomize validation
+	kubectl apply -k "${CANAL_DIR}/overlays/${CANAL_VER}" --validate=false
+
 }
 
 function metrics() {
-	kubectl apply -f 1-core-metrics/
+	METRICS_VER=${1:-v0.3.3}
+	METRICS_URL="https://github.com/kubernetes-incubator/metrics-server.git"
+	METRICS_DIR="1-core-metrics"
+	get_repo "${METRICS_URL}" "${METRICS_DIR}/overlays/${METRICS_VER}"
+	set_repo_version "${METRICS_VER}" "${METRICS_DIR}/overlays/${METRICS_VER}/metrics-server"
+	kubectl apply -k "${METRICS_DIR}/overlays/${METRICS_VER}"
+
+}
+
+function create_pvc() {
+	kubectl apply -f - <<HERE
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: test-pv-claim
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Mi
+HERE
+
 }
 
 function storage() {
 	#Start rook before any other component that requires storage
-	ROOK_URL=7-rook
-	kubectl apply -f ${ROOK_URL}/000-operator.yaml
-	while [[ $(kubectl get crd clusters.ceph.rook.io pools.ceph.rook.io >/dev/null 2>&1) || $? -ne 0 ]]; do
-		echo "Waiting for Rook CRDs"
-		sleep 2
+	ROOK_VER="${1:-v1.0.3}"
+	ROOK_URL="https://github.com/rook/rook.git"
+	ROOK_DIR=7-rook
+	get_repo "${ROOK_URL}" "${ROOK_DIR}/overlays/${ROOK_VER}"
+	set_repo_version "${ROOK_VER}" "${ROOK_DIR}/overlays/${ROOK_VER}/rook"
+	kubectl apply -k "${ROOK_DIR}/overlays/${ROOK_VER}"
+	# wait for the rook OSDs to run which means rooks should be ready
+	while [[ $(kubectl get po --all-namespaces | grep -e 'osd.*Running.*' -c) -lt 3 ]]; do
```

Comment:
When i am doing a single node setup, would this still be 3? just tested it and there are no more than 2 osds coming up. I think rook is resilient enough that we should be able to get going when we see the first osd pop up.

Reply:
I will check out single node approach.

Reply:
This works and I have updated the wait to be 1 OSD.